app/models/event.rb

```diff
@@ -3,9 +3,12 @@ class Event < ActiveRecord::Base
   belongs_to :student, foreign_key: :user_id
   belongs_to :activity
   belongs_to :creator, class_name: "Teacher"
-  has_many :scans, dependent: :destroy
 
   def scanned_in?
-    !scans.where(correct: true).empty?
+    scans.where('timestamp > ? AND timestamp < ?', start_time, end_time).present?
+  end
+
+  def scans
+    Scan.where(correct: true, student:student, location: location)
```

Comment:
So I see what's happening here and it looks good, but what is the method that changes the correct boolean attribute on scans to true or false? I looked on the Student and Scan models and I don't see it there. I'm sure I'm just not looking in the right spot. Also I noticed that scans still have an event_id attribute on the model. We want to remove that as well, correct? @jedschneider